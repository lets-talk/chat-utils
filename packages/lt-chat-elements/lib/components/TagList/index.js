import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Tag from '../Tag';
import './index.scss';

class TagList extends Component {
  onClick(item, i, e) {
    if (this.props.clickItem instanceof Function) {
      this.props.clickItem(item, i, e);
    }
  }

  render() {
    const { tags } = this.props;
    return (
      <div className="cbox-tags">
        {tags.map((tag) => (
          <Tag key={tag.name} tag={tag} onClick={this.onCLick} />
        ))}
      </div>
    );
  }
}

TagList.propTypes = {
  /**
   * tags is an array of tag objects
   * that will be displayed in the component
   */
  tags: PropTypes.array,
  /**
   * Callback function to be called when an item is clicked
   */
  clickItem: PropTypes.func,
};

TagList.defaultProps = {
  tags: [],
  clickItem: null,
};

export default TagList;
