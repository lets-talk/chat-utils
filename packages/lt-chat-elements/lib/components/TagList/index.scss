@import "./config.scss";

.cbox-tags {
  order: 2;
  margin: 5px;
}
