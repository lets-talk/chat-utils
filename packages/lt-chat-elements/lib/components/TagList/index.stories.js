import React from 'react';
// Storybook stuff
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, select, object } from '@storybook/addon-knobs';

import { WrapWithTheme, defaultTheme, themeOptions } from '../../utils/stories';
// Component to show on storybook
import { TagList } from '../../../lib';

const stories = storiesOf('TagList', module);
stories.addDecorator(withKnobs);


const tags = [
  { name: 'Tag1 Name' },
  { name: 'Tag2 Name' },
];

stories.addWithInfo(
  'Basic Usage - TagList',
  'This is the basic usage of a TagList.',
  () => (
    <div
      style={{
        position: 'fixed', bottom: 0, right: '10px', width: 'calc(100% - 20px)', height: '100%',
      }}
    >
      <WrapWithTheme
        themeName={select('Theme', themeOptions, defaultTheme)}
      >
        <TagList
          tags={object('tags', tags)}
          clickItem={action('clickItem')}
        />
      </WrapWithTheme>
    </div>
  )
);
