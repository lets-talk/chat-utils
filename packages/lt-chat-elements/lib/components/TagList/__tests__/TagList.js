import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TagList from '../index';

describe('TagList component', () => {
  const tags = [
    {
      name: 'SPU-CLAVE',
    },
  ];

  it('should render with empty props', () => {
    const component = shallow(<TagList />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with custom props', () => {
    const props = {
      tags,
      clickItem: null,
    };

    const component = shallow(<TagList {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with no tags the empty state', () => {
    const props = {
      tags: [],
      clickItem: null,
    };

    const component = shallow(<TagList {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
