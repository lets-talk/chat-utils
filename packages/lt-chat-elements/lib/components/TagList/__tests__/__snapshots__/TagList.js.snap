// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagList component should render with custom props 1`] = `
<TagListContainer>
  <Tag
    className="LT-Tag-Container"
    data={
      Object {
        "name": "SPU-CLAVE",
      }
    }
    key="SPU-CLAVE"
    onClick={null}
  />
</TagListContainer>
`;

exports[`TagList component should render with empty props 1`] = `<TagListContainer />`;

exports[`TagList component should render with no tags the empty state 1`] = `<TagListContainer />`;
